src/explore-education-statistics-common/src/components/form/__tests__/FormFieldDateInput.test.tsx

```diff
@@ -1,67 +1,70 @@
-import { Form } from '@common/components/form';
+import FormProvider from '@common/components/form/FormProvider';
+import Form from '@common/components/form/Form';
 import FormFieldDateInput from '@common/components/form/FormFieldDateInput';
 import { PartialDate } from '@common/utils/date/partialDate';
-import { render, screen, within } from '@testing-library/react';
+import Yup from '@common/validation/yup';
+import { render, screen, waitFor, within } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { Formik } from 'formik';
 import noop from 'lodash/noop';
 import React from 'react';
 
 describe('FormFieldDateInput', () => {
   test('renders correctly', () => {
     const { container } = render(
-      <Formik initialValues={{}} onSubmit={noop}>
+      <FormProvider initialValues={{}}>
         <FormFieldDateInput
           legend="Start date"
           id="startDate"
           name="startDate"
         />
-      </Formik>,
+      </FormProvider>,
     );
 
     expect(container.innerHTML).toMatchSnapshot();
   });
 
-  test('renders with an error message correctly', () => {
+  test('renders an error message correctly', async () => {
     render(
-      <Formik
+      <FormProvider
         initialValues={{}}
-        initialErrors={{
-          startDate: 'The date is wrong',
-        }}
-        initialTouched={{
-          startDate: true,
-        }}
-        onSubmit={noop}
+        validationSchema={Yup.object({
+          startDate: Yup.date().required('Select a date'),
+        })}
       >
         <FormFieldDateInput
           legend="Start date"
           id="startDate"
           name="startDate"
         />
-      </Formik>,
+      </FormProvider>,
     );
 
+    await userEvent.tab();
```

Comment:
Is there any reason we haven't used { user } = userEvent.setup() here?

Reply:
I guess to avoid making hundreds of changes to what I now realise is a mahoosive file - don't worry about it :)